src/flagpole/conditions.py

```diff
@@ -48,20 +48,20 @@ def create_case_insensitive_set_from_list(values: list[T]) -> set[T]:
     return case_insensitive_set
 
 
-class ConditionBase(BaseModel):
-    property: str = Field(description="The evaluation context property to match against.")
+@dataclasses.dataclass(frozen=True)
+class ConditionBase:
+    property: str
     """The evaluation context property to match against."""
 
-    operator: ConditionOperatorKind = Field(
-        description="The operator to use when comparing the evaluation context property to the condition's value."
-    )
-    """The operator to use when comparing the evaluation context property to the condition's value."""
-
-    value: Any = Field(
-        description="The value to compare against the condition's evaluation context property."
-    )
+    value: Any
     """The value to compare against the condition's evaluation context property."""
 
+    operator: str = dataclasses.field(default="")
```

Comment:
I don't love this, but it is less gross than having to always provide `operator` to every condition constructor throughout tests.